### PR TITLE
Nisha | Test | Version 1 Item wise export report

### PR DESCRIPTION
**Pull request by nisha-walkover**
https://app.clickup.com/t/86cv8daay

### CODE REVIEW

File: apps/web-giddh/src/app/new-inventory/component/new-inventory-advance-search/new-inventory-advance-search.component.ts [hunk 1]
```diff
@@ -84,9 +84,10 @@ export class NewInventoryAdvanceSearch implements OnInit {
             this.isMobileScreen = result.matches;
         });
         this.reportType = this.inputData?.reportType;
-        if (this.inputData?.stockReportRequest) {
+        if (this.inputData?.stockReportRequest || this.inputData?.stockReportRequestExport) {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/new-inventory/component/new-inventory-advance-search/new-inventory-advance-search.component.ts]: check if it is same interface then use  this.inputData?.stockReportRequest

File: apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts [hunk 2]
```diff
@@ -40,6 +40,8 @@ export class ReportsComponent implements OnInit {
     private destroyed$: ReplaySubject<boolean> = new ReplaySubject(1);
     /** Stock Transactional Object */
     public stockReportRequest: InventoryReportRequest = new InventoryReportRequest();
+    /** Stock Transactional Export Object */
+    public stockReportRequestExport: InventoryReportRequestExport = new InventoryReportRequestExport();
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts:46]: Correct this documentation

File: apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts [hunk 3]
```diff
@@ -597,6 +600,56 @@ export class ReportsComponent implements OnInit {
                 this.headerColumns[key].colSpan = colSpan;
             }
         });
+
+        /* This will use for table header from customise columns for export table */
+        this.stockReportRequestExport.showStockName = false;
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts]:
```suggestion
        this.stockReportRequestExport = {
        showStockName = false
        }
```

File: apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts [hunk 4]
```diff
@@ -597,6 +600,56 @@ export class ReportsComponent implements OnInit {
...
+        this.stockReportRequestExport.showOutwardsQty = false;
+        this.stockReportRequestExport.showOutwardsValue = false;
+        this.stockReportRequestExport.showClosingStockQty = false;
+        this.stockReportRequestExport.showClosingStockValue = false;
+
+        /* for column value filter selected */
+        this.displayedColumns.forEach(column => {
+            if (column === 'stock_name') {
+                this.stockReportRequestExport.showStockName = true;
+            }
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts:630]: format in else if

File: apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts [hunk 5]
```diff
@@ -729,6 +792,43 @@ export class ReportsComponent implements OnInit {
         this.router.navigate(['/pages/inventory/v2', 'stock', this.moduleType?.toLowerCase(), 'edit', element?.stock?.uniqueName], { queryParams: { tab: 1 } });
     }
 
+    /**
+     * This will use for export stock transactions report data
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts]: correct this documentation

File: apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts [hunk 6]
```diff
@@ -729,6 +792,43 @@ export class ReportsComponent implements OnInit {
...
+     *
+     * @return {*}  {void}
+     * @memberof ReportsComponent
+    */
+    public exportReport(): void {
+        setTimeout(() => {
+            this.isLoading = true;
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts]: what is use for settimeout?

File: apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts [hunk 7]
```diff
@@ -729,6 +792,43 @@ export class ReportsComponent implements OnInit {
...
+                let queryParams = {
+                    from: this.fromDate,
+                    to: this.toDate,
+                };
+                delete stockReportRequestExport.from;
+                delete stockReportRequestExport.to;
+
+                stockReportRequestExport.inventoryType = this.moduleType;
+                this.inventoryService.getItemWiseReportExport(queryParams, stockReportRequestExport).pipe(takeUntil(this.cancelApi$)).subscribe(response => {
+                    this.isLoading = true;
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts]: use component store if you are doing new task, do not use service in component basis

File: apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts [hunk 8]
```diff
@@ -729,6 +792,43 @@ export class ReportsComponent implements OnInit {
...
+                    if (response?.status === 'success') {
+                        if (typeof response?.body === "string") {
+                            this.toaster.successToast("success", response?.body);
+                            this.router.navigate(["/pages/downloads"]);
+                            this.isLoading = false;
+                        }
+                    } else {
+                        this.toaster.errorToast(response?.message);
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts]: Make is Loading false if error is coming

File: apps/web-giddh/src/app/services/apiurls/inventory.api.ts [hunk 9]
```diff
@@ -104,4 +104,5 @@ export const INVENTORY_API = {
     CREATE_DISCOUNT: 'company/:companyUniqueName/customer-vendor-discount/stock/:stockUniqueName/assign-discount', // POST Call
     UPDATE_DISCOUNT: 'company/:companyUniqueName/customer-vendor-discount/stock/:stockUniqueName/variant/:variantUniqueName/update-discount', // Patch Call
     GET_STOCK_DETAILS: 'v2/company/:companyUniqueName/stock/:stockUniqueName/details?userType=:userType', // GET call
+    INVENTORY_ITEM_WISE_EXPORT: 'company/:companyUniqueName/inventory/export-balance-report/item-wise?from=:from&to=:to',
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/services/apiurls/inventory.api.ts]:
```suggestion
    INVENTORY_ITEM_WISE_EXPORT: 'company/:companyUniqueName/inventory/export-balance-report/item-wise?from=:from&to=:to'
```

File: apps/web-giddh/src/app/new-inventory/component/reports/reports.component.html [hunk 10]
```diff
@@ -327,6 +328,9 @@ <h2 class="float-left">
             </div>
             <div class="row" *ngIf="isDataAvailable">
                 <div class="col-12 horizontal-padding">
+                    <div class="d-flex justify-content-end mt-3 mb-2" *ngIf="reportType === 'STOCK'">
+                        <button mat-stroked-button color="primary" (click)="exportReport()">Export</button>
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/new-inventory/component/reports/reports.component.html]: Add translation

File: apps/web-giddh/src/app/new-inventory/component/report-filters/report-filters.component.ts [hunk 11]
```diff
@@ -288,7 +290,14 @@ export class ReportFiltersComponent implements OnInit, OnChanges, OnDestroy {
             if (changes?.stockReportRequest?.currentValue?.warehouseUniqueNames?.length) {
                 this.selectedWarehouse = changes?.stockReportRequest?.currentValue?.warehouseUniqueNames;
                 this.stockReportRequest.warehouseUniqueNames = this.selectedWarehouse;
-                this.balanceStockReportRequest.warehouseUniqueNames = this.selectedWarehouse;
+            }
+            if (changes?.stockReportRequestExport?.currentValue?.branchUniqueNames?.length) {
+                this.selectedBranch = changes?.stockReportRequestExport?.currentValue?.branchUniqueNames;
+                this.stockReportRequestExport.branchUniqueNames = this.selectedBranch;
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/new-inventory/component/report-filters/report-filters.component.ts]: use stockReportRequest if interface is same and assign value into your interface from this interface.

File: apps/web-giddh/src/app/new-inventory/component/report-filters/report-filters.component.ts [hunk 12]
```diff
@@ -42,6 +42,8 @@ export class ReportFiltersComponent implements OnInit, OnChanges, OnDestroy {
     @Input() public fromToDate: any = {};
     /** Stock Transactional Object */
     @Input() public stockReportRequest: StockTransactionReportRequest = new StockTransactionReportRequest();
+    /** Stock Transactional Export Object */
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/new-inventory/component/report-filters/report-filters.component.ts]: Correct this documentation

**Review by dilpreetsinghofficial — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts (hunk 2 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts:46]: done

File: apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts (hunk 3 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts (hunk 4 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts:630]: done

File: apps/web-giddh/src/app/models/api-models/Inventory.ts [hunk 13]
```diff
@@ -448,6 +498,52 @@ export class InventoryReportRequest {
...
+    public expression?: string;
+    public val?: number;
+    public stockGroupUniqueNames?: any[];
+    public stockUniqueNames?: any[];
+    public warehouseUniqueNames?: any[];
+    public variantUniqueNames?: any[];
+    public from: string = '';
+    public to: string = '';
+    public inventoryType?: string;
+    results: any;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/models/api-models/Inventory.ts]:
```suggestion
    public results: any = any[];
```

File: apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts (hunk 5 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts]: done

File: apps/web-giddh/src/app/models/api-models/Inventory.ts [hunk 14]
```diff
@@ -448,6 +498,52 @@ export class InventoryReportRequest {
...
+    toDate: any;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/models/api-models/Inventory.ts]:
```suggestion
   public toDate: string = ''
```

File: apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts (hunk 6 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts (hunk 8 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts]: done

File: apps/web-giddh/src/app/services/apiurls/inventory.api.ts (hunk 9 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/services/apiurls/inventory.api.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/reports/reports.component.html (hunk 10 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/reports/reports.component.html]: done

File: apps/web-giddh/src/app/new-inventory/component/report-filters/report-filters.component.ts (hunk 12 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/report-filters/report-filters.component.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts [hunk 15]
```diff
@@ -729,6 +792,43 @@ export class ReportsComponent implements OnInit {
...
+                };
+                delete stockReportRequestExport.from;
+                delete stockReportRequestExport.to;
+
+                stockReportRequestExport.inventoryType = this.moduleType;
+                this.inventoryService.getItemWiseReportExport(queryParams, stockReportRequestExport).pipe(takeUntil(this.cancelApi$)).subscribe(response => {
+                    this.isLoading = true;
+                    if (response?.status === 'success') {
+                        if (typeof response?.body === "string") {
+                            this.toaster.successToast("success", response?.body);
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts]:
```suggestion
                            this.toaster.showSnackBar("success", response?.body);
```

File: apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts (hunk 8 above)
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts]:
```suggestion
                        this.toaster. showSnackBar("error",response?.message);
```

File: apps/web-giddh/src/app/services/inventory.service.ts [hunk 16]
```diff
@@ -1290,6 +1291,30 @@ export class InventoryService {
...
+     * @param {*} queryParams
+     * @param {InventoryReportRequestExport} stockReportRequest
+     * @return {*}  {Observable<BaseResponse<InventoryReportRequestExport, InventoryReportRequest>>}
+     * @memberof InventoryService
+     */
+    public getItemWiseReportExport(queryParams: any, stockReportRequest: InventoryReportRequestExport): Observable<BaseResponse<InventoryReportRequestExport, InventoryReportRequest>> {
+        this.companyUniqueName = this.generalService.companyUniqueName;
+        return this.http.post(this.config.apiUrl + INVENTORY_API.INVENTORY_ITEM_WISE_EXPORT
+            ?.replace(':companyUniqueName', encodeURIComponent(this.companyUniqueName))
+            ?.replace(':from', encodeURIComponent(queryParams.from))
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/services/inventory.service.ts]:
```suggestion
            ?.replace(':from', encodeURIComponent(queryParams?.from))
```

File: apps/web-giddh/src/app/services/inventory.service.ts [hunk 17]
```diff
@@ -1290,6 +1291,30 @@ export class InventoryService {
...
+            ?.replace(':to', encodeURIComponent(queryParams.to))
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/services/inventory.service.ts]:
```suggestion
            ?.replace(':to', encodeURIComponent(queryParams?.to))
```

**Review by dvCodeWorld — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts (hunk 8 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts (hunk 15 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts]: done

File: apps/web-giddh/src/app/services/inventory.service.ts (hunk 16 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/services/inventory.service.ts]: done

File: apps/web-giddh/src/app/services/inventory.service.ts (hunk 17 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/services/inventory.service.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/new-inventory-advance-search/new-inventory-advance-search.component.ts (hunk 1 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/new-inventory-advance-search/new-inventory-advance-search.component.ts]: done

File: apps/web-giddh/src/app/models/api-models/Inventory.ts (hunk 13 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/models/api-models/Inventory.ts]: done

**Review by dvCodeWorld — APPROVED** [review]

File: apps/web-giddh/src/app/new-inventory/component/report-filters/report-filters.component.ts (hunk 11 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/report-filters/report-filters.component.ts]: done

**Review by dilpreetsinghofficial — APPROVED** [review]

File: apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts [hunk 18]
```diff
@@ -729,6 +795,42 @@ export class ReportsComponent implements OnInit {
...
     }
 
+    /**
+     * This will use for export stock report data
+     *
+     * @return {*}  {void}
+     * @memberof ReportsComponent
+    */
+    public exportReport(): void {
+        this.isLoading = true;
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts]: move this in if condition

File: apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts [hunk 19]
```diff
@@ -729,6 +795,42 @@ export class ReportsComponent implements OnInit {
...
+            let queryParams = {
+                from: this.fromDate,
+                to: this.toDate,
+            };
+            delete stockReportRequestExport.from;
+            delete stockReportRequestExport.to;
+
+            stockReportRequestExport.inventoryType = this.moduleType;
+            this.inventoryService.getItemWiseReportExport(queryParams, stockReportRequestExport).pipe(takeUntil(this.cancelApi$)).subscribe(response => {
+                this.isLoading = true;
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts]: remove this

File: apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts [hunk 20]
```diff
@@ -729,6 +795,42 @@ export class ReportsComponent implements OnInit {
...
+            let stockReportRequestExport = this.stockReportRequestExport;
+            let queryParams = {
+                from: this.fromDate,
+                to: this.toDate,
+            };
+            delete stockReportRequestExport.from;
+            delete stockReportRequestExport.to;
+
+            stockReportRequestExport.inventoryType = this.moduleType;
+            this.inventoryService.getItemWiseReportExport(queryParams, stockReportRequestExport).pipe(takeUntil(this.cancelApi$)).subscribe(response => {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts]: why are we not using component store??

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts (hunk 18 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/inventory.store.ts [hunk 21]
```diff
@@ -0,0 +1,48 @@
...
+        super({});
+    }
+
+    readonly exportStock = this.effect((data$: Observable<{ stockReportRequest: any, queryParams: any }>) => {
+        return data$.pipe(
+            switchMap((req) => {
+                return this.inventoryService.getItemWiseReportExport(req.queryParams, req.stockReportRequest).pipe(
+                    tapResponse(
+                        (res: BaseResponse<any, any>) => {
+                            console.log(res)
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/inventory.store.ts]: remove it

File: apps/web-giddh/src/app/new-inventory/component/inventory.store.ts [hunk 22]
```diff
@@ -0,0 +1,48 @@
...
+                            if (res.status === "success") {
+                                if (typeof res?.body === "string") {
+                                    this.toaster.showSnackBar("success", res.body);
+                                    this.router.navigate(["/pages/downloads"]);
+                                }
+                            } else {
+                                this.toaster.showSnackBar("error", res.message);
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/inventory.store.ts]:
```suggestion
                                this.toaster.showSnackBar("error", res?.message);
```

**Review by dvCodeWorld — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/new-inventory/component/inventory.store.ts [hunk 23]
```diff
@@ -0,0 +1,48 @@
...
+    constructor(
+        private store: Store<AppState>,
+        private inventoryService: InventoryService,
+        private toaster: ToasterService,
+        public router: Router
+    ) {
+        super({});
+    }
+
+    readonly exportStock = this.effect((data$: Observable<{ stockReportRequest: any, queryParams: any }>) => {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/new-inventory/component/inventory.store.ts:38]: Add documentation

File: apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts [hunk 24]
```diff
@@ -729,6 +799,33 @@ export class ReportsComponent implements OnInit {
...
+     * @memberof ReportsComponent
+    */
+    public exportReport(): void {
+        if (this.reportType === InventoryReportType.stock) {
+            this.isLoading = true;
+
+            let stockReportRequestExport = this.stockReportRequestExport;
+            let queryParams = {
+                from: this.fromDate,
+                to: this.toDate,
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts]:
```suggestion
                to: this.toDate
```

File: apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts [hunk 25]
```diff
@@ -729,6 +799,33 @@ export class ReportsComponent implements OnInit {
...
+            };
+            delete stockReportRequestExport.from;
+            delete stockReportRequestExport.to;
+
+            stockReportRequestExport.inventoryType = this.moduleType;
+            // data is coming from inventory store
+            this.componentStore.exportStock({
+                stockReportRequest: stockReportRequestExport,
+                queryParams: queryParams,
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts]:
```suggestion
                queryParams: queryParams
```

**Review by dilpreetsinghofficial — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/new-inventory/component/inventory.store.ts (hunk 22 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/inventory.store.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/inventory.store.ts (hunk 23 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/inventory.store.ts:38]: done

File: apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts (hunk 24 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts (hunk 25 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/inventory.store.ts (hunk 21 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/inventory.store.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts (hunk 7 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts (hunk 19 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts (hunk 20 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/reports/reports.component.ts]: Now using Store

**Review by dvCodeWorld — APPROVED** [review]

**Review by dilpreetsinghofficial — APPROVED** [review]

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**